### DISCUSSION TITLE
upgrade pnpm via home-brew installation

### DISCUSSION THREAD

**Discussion by ciarantr** [pnpm/pnpm#4622] [Q&A] [answered]
HI all, I installed pnpm using homebrew && I'm having an issue when I upgrade pnpm to the latest version pnpm does not seem to upgrade as well.

Any advice would be appreciated.

**Comment by zkochan** [comment]
@umireon maybe you could help.

**Reply by rwieruch** [reply]
Ran into the same problem. brew upgrade pnpm yields 7.1.5. But when I check with pnpm -v I get the old version 6.11.0.

Even a brew uninstall pnpm and brew install pnpm gives yields the same result.

**Comment by standingdreams** [comment]
Any update on how to handle this? OR should we just remove PNPM from brew and install with npm?

**Reply by zkochan** [reply]
I have no idea as I don't own a mac. If @umireon can't help or someone else, then I would suggest using the standalone script to install pnpm, corepack, or npm. The standalone script is the best choice because then you may also use pnpm to manage node.js versions.

All the installation methods are here: https://pnpm.io/installation

**Comment by ciarantr** [comment, marked as answer]
I solved the issue by reinstalling pnpm. No issues updating pnpm using Homebrew.

**Reply by rwieruch** [reply]
`brew uninstall pnpm` and `brew install pnpm` still yields the old version for me.

**Reply by mchingiz** [reply]
`brew link` was failing, so running these two before reinstalling helped
`rm '/opt/homebrew/bin/pnpm'`
`rm '/opt/homebrew/bin/pnpx'`

**Reply by ivarrreyna** [reply]
This worked for me

**Reply by bongudth** [reply]
This worked for me
```
brew uninstall pnpm
brew install pnpm
```

**Comment by RileyManda** [comment]
I was facing the same issue and after conducting some research:I managed to  update pnpm as follows:

**version before update:** 👇🏻 :
`pnpm -v`

<img width="292" alt="Screen Shot 2023-08-28 at 10 37 39 PM" src="https://github.com/pnpm/pnpm/assets/37291194/3f87ebb8-ac01-43ee-af5d-e98eeea8eaac">

Update pnpm:👇🏻 :

`pnpm add -g pnpm`

<img width="659" alt="Screen Shot 2023-08-28 at 10 38 33 PM" src="https://github.com/pnpm/pnpm/assets/37291194/a80c9919-0335-40db-a9be-9234d9b74a02">

pnpm updated: 👇🏻 :

<img width="428" alt="Screen Shot 2023-08-28 at 10 39 02 PM" src="https://github.com/pnpm/pnpm/assets/37291194/bcf8ba3c-c3f8-4525-b306-fca08e246a02">

**Reply by cesarm-meyersound** [reply]
I don't believe you're using homebrew here. When installed with homebrew the installiation path will look like 
```sh
where pnpm
 /opt/homebrew/bin/pnpm
```
Not /Users/...

**Comment by shalithadev** [comment]
`pnpm add -g pnpm`

If ERR_PNPM_NO_GLOBAL_BIN_DIR  Unable to find the global bin directory

`pnpm setup`

`source /Users/<user>/.zshrc`

`pnpm add -g pnpm`

`source /Users/<user>/.zshrc`

`pnpm -v`

**Comment by ryenus** [comment]
When using [corepack from Node][1] (since v16.13), run below command:

```sh
corepack install -g pnpm@latest
```

See also: https://nodejs.org/api/corepack.html#upgrading-the-global-versions

[1]: https://pnpm.io/installation#using-corepack

**Reply by seamusriordan** [reply]
In case anyone runs across this and the above doesn't work, I found that I had set the version of pnpm in my `package.json` for cicd reasons

```
  "packageManager": "pnpm@9.15.4"
```

Upgrading and then changing that line to the desired version resolved the issue.

**Reply by ryenus** [reply]
You can use [corepack use](https://github.com/nodejs/corepack#corepack-use-nameversion) to ask Corepack to update your local package.json to use the package manager of your choice:

> `corepack use pnpm@7.x` # sets the latest 7.x version in the package.json
> `corepack use yarn@*` # sets the latest version in the package.json

See https://nodejs.org/api/corepack.html#corepack_configuring_a_package

**Comment by samtigall** [comment]
`pnpm self-update`
did it for me on mac for v9 -> v10

**Comment by Sam-Kruglov** [comment]
Turns out I installed via brew but then it looks like that after I ran `pnpm self-update` it actually installed a second pnpm somewhere else. So it kept conflicting. After I did `brew uninstall pnpm`, it stabilized.
